pgdog/src/backend/pool/cluster.rs

```diff
@@ -60,8 +63,26 @@ impl ShardingSchema {
     pub fn tables(&self) -> &ShardedTables {
         &self.tables
     }
+
+    pub fn has_global_sharding_key_with_mappings(&self) -> bool {
+        let mut global_sharding_rules_with_mappings = 0;
+        let mut other_rules = 0;
+
+        for table in self.tables.tables() {
+            if table.name.is_none() && table.mapping.is_some() {
+                global_sharding_rules_with_mappings += 1;
+            } else {
+                other_rules += 1;
+            }
+        }
+
+        global_sharding_rules_with_mappings == 1 && other_rules == 0
+    }
 }
 
+// -------------------------------------------------------------------------------------------------
```

Comment:
Checkout the Rust style guide for comments: https://doc.rust-lang.org/book/ch03-04-comments.html

Could be cool to use it so we can generate documentation automatically using `rust doc`.